plugins/pulp_docker/plugins/models.py

```diff
@@ -248,21 +249,28 @@ class ManifestList(pulp_models.FileContentUnit, UnitMixin):
     @classmethod
     def from_json(cls, manifest_json, digest):
         """
-        Construct and return a DockerManifestList from the given JSON document.
+        Construct and return a Docker ManifestList from the given JSON document.
 
-        :param manifest_json: A JSON document describing a DockerManifest object as defined by the
+        :param manifest_json: A JSON document describing a ManifestList object as defined by the
                               Docker v2, Schema 2 Manifest List documentation.
         :type  manifest_json: basestring
         :param digest:        The content digest of the manifest, as described at
                               https://docs.docker.com/registry/spec/api/#content-digests
         :type  digest:        basestring
 
-        :return:              An initialized DockerManifestList object
-        :rtype:               pulp_docker.common.models.DockerManifest
+        :return:              An initialized ManifestList object
+        :rtype:               pulp_docker.plugins.models.DockerManifest
         """
 
-        mediatype = 'application/vnd.docker.distribution.manifest.v2+json'
-        manifest = json.loads(manifest_json)
+        try:
+            manifest = json.loads(manifest_json)
+        except ValueError:
+            raise PulpCodedValidationException(error_code=error_codes.DKR1011)
+
+        if manifest.get('mediaType') != constants.MEDIA_TYPE_MANIFEST_LIST:
```

Comment:
i would make this kind of checks after from_json().
reasons behind - This method is used in sync, therefore the checks would be applied as well, but in fact they are surplus here, because registry did some sanity checks when client tried to push the content to the registry, therefore we should not bother to validate this because registry already did it.

Reply:
I had considered adding this check to `from_json()`, but I didn't want to break anything in sync. I agree that this check isn't strictly necessary, but given that the user will manually change the file, it makes sense. I did this check before `from_json` to fail fast. If the upload isn't the right media, I think we should check, especially since it is easy. I did this before  `from_json` to fail fast.